src/lib.rs

```diff
@@ -495,8 +494,11 @@ pub enum ApiError {
     #[error("this account is not authorized for this action")]
     Forbidden,
 
-    #[error("{0}")]
-    InvalidModification(String),
+    #[error("must include the --replace flag to update an existing release")]
```

Comment:
This error talks about a command line interface, but this library does not have one. Perhaps we should talk about a replace parameter instead of a flag.

Reply:
That makes sense, but there is a problem. `gleam-cli` just outputs whatever error message this library returns. So if we change this, we either have to output a message that does not mention the flag, which will be confusing to the developer and they would need to figure out the flag by themselves. I know, they can just check `gleam help publish` but it's not ideal that the error message is not enough!
Another solution is that `gleam-cli` does not just output the error message given by this library and actually checks for the specific error and outputs the proper error message. But if we do that, then why do we even have an error message here? And it also doesn't feel right that we always just output whatever this library returns except for that one case!

Reply:
If there is something that the user can do we should specifically handle that error variant and show an appropriate error message. We do this a few times in the compiler.

We don't want this rust library to talk about a command line interface because it would be incorrect any time it is used in something that is not a command line interface or does not support that flag.

Thank you